### PR TITLE
Fix the list_priorities script to work again.

### PR DESCRIPTION
**Pull request by clalancette**
In particular, we needed to make sure that we support !include
in YAML, which is new since the last time this script was
touched (in 2016!).  Along with that, revamp the whole
structure of it, since we now have to be more careful about
directories.  And finally, make it very specifically look at
particular directories, since nowadays not all directories
contain jenkins config scripts.

With these changes, it can now be used again to list out the
priorities.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

**Review by ivanpauno — APPROVED** [review]
This probably needs to be reviewed by someone with more experience using this script, but codewise lgtm

**Comment by clalancette** [comment]
> This probably needs to be reviewed by someone with more experience using this script, but codewise lgtm

This script is only used by hand to verify the priorities.  So with that said, I don't think it is a big deal if we were to slightly break it or whatever.

That said, this same script also exists in https://github.com/ros-infrastructure/ros_buildfarm_config .  @nuclearsandwich @cottsay given that these two repositories are very far apart in terms of configuration, would you be OK with me merging this and then copying it over to that repository?  For that matter, I think it might make sense to break the link between the two repositories, but I'd like to hear your opinions about that first.

**Comment by cottsay** [comment]
@nuclearsandwich and I have discussed preparing the `ros_buildfarm` repository for distribution in deb form. We should consider moving this script there, which is where the config check script running this repo's CI lives already.

Then we can use `ros_buildfarm`'s !include handler and drop the new dep. Or even inspect the parsed config objects instead of parsing the yaml manually.

**Review by cottsay — APPROVED** [review]
That said, we're probably not ready for that yet, and this improves the current situation for the time being.

File: list_priorities.py [hunk 1]
```diff
@@ -1,47 +1,48 @@
...
+import yamlinclude
 
-def main():
-    priorities = get_priorities()
-    for key in sorted(priorities.keys()):
-        print(key)
-        for value in sorted(priorities[key]):
-            print(' ', value)
 
+DIRECTORIES = ['foxy', 'galactic', 'humble', 'rolling']
```
Comment (nuclearsandwich) [inline list_priorities.py:10]: If we're going to do this we should really be parsing the index to extract the rosdistro names and use them as directories, or even fetch the target YAML files directly.

**Review by nuclearsandwich — COMMENTED** [review]
I like the restructuring that this PR provides but I do need to affect a certain amount of insult on behalf of https://github.com/ros2/ros_buildfarm_config/pull/135 which has been languishing since October 2020 and was recently updated with a stub implementation of includes that requires no additional dependency

**Comment by clalancette** [comment]
> I like the restructuring that this PR provides but I do need to affect a certain amount of insult on behalf of https://github.com/ros2/ros_buildfarm_config/pull/135 which has been languishing since October 2020 and was recently updated with a stub implementation of includes that requires no additional dependency

Apologies, I didn't even look at other open PRs.  I should have done that.

File: list_priorities.py (hunk 1 above)
Comment (clalancette) [reply list_priorities.py:10]: 100% agreed; I didn't even think of that.  See https://github.com/ros2/ros_buildfarm_config/pull/234 , which implements this.

**Pull request merged**